lib/Dialect/AIE/IR/AIEDialect.cpp

```diff
@@ -546,19 +531,18 @@ LogicalResult xilinx::AIE::ObjectFifoLinkOp::verify() {
 
   if (isJoin()) {
     ObjectFifoCreateOp fifoOut = getOutputObjectFifos()[0];
-    AIEObjectFifoType fifoType =
-        fifoOut.getElemType().cast<AIEObjectFifoType>();
-    MemRefType elemType = fifoType.getElementType().cast<MemRefType>();
+    auto fifoType = fifoOut.getElemType().cast<AIEObjectFifoType>();
+    auto elemType = fifoType.getElementType().cast<MemRefType>();
     int64_t outputSize = 1;
     for (auto dim : elemType.getShape())
       outputSize *= dim;
 
     int inputSize = 0;
     for (auto fifoIn : getInputObjectFifos()) {
-      AIEObjectFifoType fifo = fifoIn.getElemType().cast<AIEObjectFifoType>();
-      MemRefType elemType = fifo.getElementType().cast<MemRefType>();
```

Comment:
we need to stop doing this in general - LLVM highly recommends spelling out types fully unless they're long template specializations.